native/packages/react-native-bpk-component-image/src/BpkImage.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  type StyleObj,
+  StyleSheet,
+  View,
+  ViewPropTypes,
+  Animated,
+} from 'react-native';
+import BpkSpinner from 'react-native-bpk-component-spinner';
+
+import {
+  borderRadiusSm,
+  colorGray300,
+  animationDurationBase,
+} from 'bpk-tokens/tokens/base.react.native';
+
+const styles = StyleSheet.create({
+  outer: {
+    width: '100%',
+    height: '100%',
+  },
+  withBorderRadius: {
+    borderRadius: borderRadiusSm,
+    overflow: 'hidden',
+  },
+  image: { width: '100%', height: '100%', position: 'absolute' },
+  spinnerView: {
+    width: '100%',
+    height: '100%',
+    backgroundColor: colorGray300,
+    justifyContent: 'center',
+    alignItems: 'center',
+    position: 'absolute',
+  },
+});
+
+export type Props = {
+  style: StyleObj,
+  inView: ?boolean,
+  borderRadius: ?boolean,
+  loaded: ?boolean,
+  alt: string,
+  source: string,
+  onLoad: ?func,
+};
+
+type State = {
+  renderSpinnerView: boolean,
```

Comment:
To me this name implies that it's going to be a function - think of `render()` for example. Maybe `showSpinnerView` instead?

Reply:
Actually, let's not use the term `spinner` because the fact that we're using a spinner is an internal implementation detail. If we changed to use something else that wasn't a spinner, we'd have to change this variable too. How about `showLoadingIndicator`?

Note: we should remove `spinner` elsewhere too like in the stylesheet naming.